week1/homework/src/server.js

```diff
@@ -9,7 +9,40 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    const reqUrl = request.url;
+    if (reqUrl === '/' || reqUrl === '/state') {
+      response.writeHead(200, { 'Content-Type': 'application/json' });
+      const home = {
+        'state': state
+      };
+      response.end(JSON.stringify(home, null, 2));
+    } else if (reqUrl === '/add') {
+      response.writeHead(200, { 'Content-Type': 'application/json' });
```

Comment:
You are having similar logic in every case of the if statement. You should movr this logic inside of a function and call the function from the different if blocks, passing the right information trough arguments.

Reply:
got it